editor.planx.uk/src/@planx/components/PlanningConstraints/List.tsx

```diff
@@ -0,0 +1,223 @@
+import Box, { BoxProps } from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Collapse from "@mui/material/Collapse";
+import Link from "@mui/material/Link";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListSubheader from "@mui/material/ListSubheader";
+import { styled } from "@mui/material/styles";
+import Typography from "@mui/material/Typography";
+import type {
+  Constraint,
+  GISResponse,
+  Metadata,
+} from "@opensystemslab/planx-core/types";
+import groupBy from "lodash/groupBy";
+import React, { ReactNode, useState } from "react";
+import ReactHtmlParser from "react-html-parser";
+import Caret from "ui/icons/Caret";
+import ReactMarkdownOrHtml from "ui/ReactMarkdownOrHtml";
+
+const CATEGORY_COLORS: Record<string, string> = {
+  "General policy": "#99C1DE",
+  "Heritage and conservation": "#EDDCD2",
+  Ecology: "#E0EFCC",
+  Trees: "#DBE7E4",
+};
+
+interface StyledConstraintProps extends BoxProps {
+  category: string;
+}
+
+const StyledConstraint = styled(Box, {
+  shouldForwardProp: (prop) => prop !== "category",
+})<StyledConstraintProps>(({ theme, category }) => ({
+  borderLeft: `5px solid ${CATEGORY_COLORS[category]}`,
+  paddingRight: 0,
+  width: "100%",
+  color: theme.palette.text.primary,
+  position: "relative",
+  "&::after": {
+    content: "''",
+    position: "absolute",
+    bottom: 0,
+    left: 0,
+    width: "100%",
+    height: "1px",
+    background: theme.palette.secondary.main,
+  },
+}));
+
+interface ConstraintsListProps {
+  data: Constraint[];
+  metadata: GISResponse["metadata"];
+}
+
+export default function ConstraintsList({
+  data,
+  metadata,
+}: ConstraintsListProps) {
+  const groupedConstraints = groupBy(data, (constraint: Constraint) => {
+    return constraint.category;
+  });
+
+  return (
+    <Box mb={3}>
+      <List dense disablePadding>
+        {Object.keys(groupedConstraints).map((category: string) => (
+          <ListSubheader
+            disableGutters
+            disableSticky
+            color="primary"
+            component="div"
+            key={category}
+            style={{
+              padding: 0,
+              backgroundColor: CATEGORY_COLORS[category],
+              marginBottom: "2em",
+            }}
+          >
+            <Typography
+              variant="body1"
+              component="h3"
+              py={1}
+              px={2}
+              pl={2.5}
+              style={{
+                fontWeight: 700,
+                color: "black",
+              }}
+            >
+              {category}
+            </Typography>
+            {groupedConstraints[category].map((con: any) => (
```

Comment:
This one is tricky! `text` is actually optional - `designated` is the prime example: it has a `value` so that is can be referenced in automations, but does _not_ have `text` because it should never be shown as a unique item in the list of constraints.

So what these filters are trying to say is - filter out constraints that don't have text, then filter again on value. Was trying to avoid this `any` type but was getting a whole load of red, but I'll see if I can properly represent this as I'm updating the `GISResponse` type in planx-core too :+1: